catalog-rest-service/src/main/java/org/openmetadata/catalog/jdbi3/FeedRepository.java

```diff
@@ -191,50 +188,60 @@ public ThreadCount getThreadsCount(String link, boolean isResolved) throws IOExc
-                entityLink.getFullyQualifiedFieldType(),
-                "thread",
-                Relationship.MENTIONED_IN.ordinal());
-    result.forEach(l -> threadIds.add(l.get(1)));
-
-    // TODO remove hardcoding of thread
-    // For a user entitylink get created or replied relationships to the thread
-    if (reference.getType().equals(Entity.USER)) {
-      threadIds.addAll(getUserThreadIds(reference));
+      threads = JsonUtils.readObjects(dao.feedDAO().list(), Thread.class);
     } else {
-      // Only data assets are added as about
-      result =
-          dao.fieldRelationshipDAO()
-              .listFromByAllPrefix(
-                  entityLink.getFullyQualifiedFieldValue(),
-                  "thread",
-                  entityLink.getFullyQualifiedFieldType(),
-                  Relationship.IS_ABOUT.ordinal());
-      result.forEach(l -> threadIds.add(l.get(1)));
+      EntityLink entityLink = EntityLink.parse(link);
+      EntityReference reference = EntityUtil.validateEntityLink(entityLink);
+      List<String> threadIds = new ArrayList<>();
+      List<List<String>> result;
+
+      // TODO remove hardcoding of thread
+      // For a user entitylink get created or replied relationships to the thread
+      if (reference.getType().equals(Entity.USER)) {
+        threadIds.addAll(getUserThreadIds(reference));
+      } else {
+        // Only data assets are added as about
+        result =
+            dao.fieldRelationshipDAO()
+                .listFromByAllPrefix(
+                    entityLink.getFullyQualifiedFieldValue(),
+                    "thread",
+                    entityLink.getFullyQualifiedFieldType(),
+                    Relationship.IS_ABOUT.ordinal());
+        result.forEach(l -> threadIds.add(l.get(1)));
+      }
+
+      Set<String> uniqueValues = new HashSet<>();
+      for (String t : threadIds) {
+        // If an entity has multiple relationships (created, mentioned, repliedTo etc.) to the same thread
+        // Don't send duplicated copies of the thread in response
+        if (uniqueValues.add(t)) {
+          threads.add(EntityUtil.validate(t, dao.feedDAO().findById(t), Thread.class));
+        }
+      }
+      // sort the list by thread updated timestamp before returning
+      threads.sort(Comparator.comparing(Thread::getUpdatedAt, Comparator.reverseOrder()));
     }
+    return limitPostsInThreads(threads, limitPosts);
+  }
 
-    List<Thread> threads = new ArrayList<>();
-    Set<String> uniqueValues = new HashSet<>();
-    for (String t : threadIds) {
-      // If an entity has multiple relationships (created, mentioned, repliedTo etc.) to the same thread
-      // Don't send duplicated copies of the thread in response
-      if (uniqueValues.add(t)) {
-        threads.add(EntityUtil.validate(t, dao.feedDAO().findById(t), Thread.class));
+  private List<Thread> limitPostsInThreads(List<Thread> threads, int limitPosts) {
+    if (limitPosts == 0) {
```

Comment:
If limit is 0, we return all posts?

Should we do `t.withPosts(null)` when limit = 0? As in return 0 posts per thread

Reply:
With this patch, both `t.withPosts(null)` and `t.withPosts(0)` will return all the posts in the thread. Do you think we should add support for returning just the thread with 0 posts?

Reply:
I was possibly misled when I saw that we don't return 0 posts when limit is 0.

From a UX point of view, it seems like we'll always need a non-zero number of posts to show our users. So this logic is good. 

May be add a comment to this line indicating the above, for future reference.

Reply:
Second thought:
May be also check what is the behavior with pagination and do something similar here.
What happens if we put limit 0 for pagination of GET /tables?

Reply:
For pagination:

```
      @Parameter(description = "Limit the number tables returned. (1 to 1000000, default = 10)")
          @DefaultValue("10")
          @Min(1)
          @Max(1000000)
          @QueryParam("limit")
```

So all we have to do is document the API accordingly similar to the one above (in FeedResource.java)
Indicate that if limit = 0 it returns everything. -- so this is clear to end users.

Reply:
Sure, added a comment and unit test coverage.

Reply:
Added the min and max limits in the API for limitPosts and updated the tests. Please take another look. Thank you!

Reply:
Looks much better with API docs - thanks for incorporating this (especially as I was thinking back and forth through this and providing feedback on the fly)